pkg/cmd/get/get.go

```diff
@@ -0,0 +1,277 @@
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package get
+
+import (
+	"fmt"
+	"time"
+
+	kruiseappsv1alpha1 "github.com/openkruise/kruise-api/apps/v1alpha1"
+	kruiseappsv1beta1 "github.com/openkruise/kruise-api/apps/v1beta1"
+	rolloutv1alpha1 "github.com/openkruise/kruise-rollout-api/rollouts/v1alpha1"
+	rolloutv1beta1 "github.com/openkruise/kruise-rollout-api/rollouts/v1beta1"
+	internalpolymorphichelpers "github.com/openkruise/kruise-tools/pkg/internal/polymorphichelpers"
+	"github.com/spf13/cobra"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/cli-runtime/pkg/genericclioptions"
+	"k8s.io/cli-runtime/pkg/resource"
+	cmdutil "k8s.io/kubectl/pkg/cmd/util"
+	"k8s.io/kubectl/pkg/scheme"
+	"k8s.io/kubectl/pkg/util/i18n"
+	"k8s.io/kubectl/pkg/util/templates"
+)
+
+var (
+	getLong = templates.Examples(i18n.T(`
+		Display one or many resources related to kruise.`))
+
+	getExample = templates.Examples(i18n.T(`
+		# List all resources in the default namespace
+		kubectl-kruise get all 
+
+		# List all resources in the specific namespace
+		kubectl-kruise get all -n namespace
+
+		# Watch all resources in the default namespace
+		kubectl-kruise get all -w`))
+)
+
+type GetOptions struct {
+	genericclioptions.IOStreams
+	Builder          func() *resource.Builder
+	Resources        []string
+	Namespace        string
+	EnforceNamespace bool
+	GetViewerFn      internalpolymorphichelpers.GetViewerFunc
+}
+
+func NewCmdGet(f cmdutil.Factory, streams genericclioptions.IOStreams) *cobra.Command {
+	o := &GetOptions{IOStreams: streams}
+
+	cmd := &cobra.Command{
+		Use:                   "get all",
+		DisableFlagsInUseLine: true,
+		Short:                 i18n.T("Display one or many resources"),
+		Long:                  getLong,
+		Example:               getExample,
+		Run: func(cmd *cobra.Command, args []string) {
+			cmdutil.CheckErr(o.Complete(f, args))
+			cmdutil.CheckErr(o.Run())
+		},
+	}
+
+	cmd.Flags().StringVarP(&o.Namespace, "namespace", "n", o.Namespace, "If present, the namespace scope for this CLI request")
+
+	return cmd
+}
+
+func (o *GetOptions) Complete(f cmdutil.Factory, args []string) error {
+	var err error
+
+	if len(args) == 0 {
+		return fmt.Errorf("required resource not specified")
+	}
+
+	o.Resources = args
+	o.Namespace, o.EnforceNamespace, err = f.ToRawKubeConfigLoader().Namespace()
+	if err != nil {
+		return err
+	}
+
+	o.GetViewerFn = internalpolymorphichelpers.GetViewerFn
+	o.Builder = f.NewBuilder
+
+	return nil
+}
+
+func (o *GetOptions) Run() error {
+	if len(o.Resources) == 0 {
+		return fmt.Errorf("you must specify the type of resource to get")
+	}
+
+	if o.Resources[0] == "all" {
+		resourceTypes := []string{
```

Comment:
Other resources such like `crr`, `broadcast job` etc not shown in `get all` cmd ?